relevanceai/vector_tools/cluster.py

```diff
@@ -627,3 +731,130 @@ def hdbscan_cluster(
+        1- Loads the data
+        2- Clusters the data
+        3- Updates the data with clustering info
+        4- Adds the centroid to the hidden centroid collection
+
+        Parameters
+        ----------
+        dataset_id : string
+            name of the dataser
+
+        vector_fields : list
+            a list containing the vector field to be used for clustering
+
+        filters : list
+            a list to filter documents of the dataset,
+
+        n_clusters : int
+            Number of clusters expected in the data
+
+        affinity : str or callable, default='euclidean'
+            The distance metric used to compute the distance between vectors in dataset
+
+        memory : Memory(cachedir=None)
+            hierarchical configuration parameter on memory management
+
+        compute_full_tree: str = 'auto' or bool
+            parameter for early stopping to avoid compute the entire dendrogram
+
+        linkage : str = 'ward'
+            specifiy the type of linkage between observations in each cluster.
+            choose from {'ward', 'complete', 'average', 'single'}
+
+        distance_threshold : float
+            The maximum distance for which clusters can be to not be merged together
+
+        compute_distances : bool
+            Computes distances between clusters even if distance_threshold is not used. 
+            This can be used to make dendrogram visualization, but introduces a computational and memory overhead.
+
+        alias : string
+            "hierarchical", string to be used in naming of the field showing the clustering results
+
+        cluster_field: string
+            "_cluster_", string to name the main cluster field
+
+        overwrite : bool
+            False by default, To overwite an existing clusering result
+
+        Example
+        -------------
+
+        >>> client.vector_tools.cluster.hierarchical_cluster(
+            dataset_id="sample_dataset",
+            vector_fields=["sample_1_vector_"] # Only 1 vector field is supported for now
+        )
+        """
+
+        if (
+            ".".join([cluster_field, vector_fields[0], alias])
+            in self.datasets.schema(dataset_id)
+            and not overwrite
+        ):
+            raise ClusteringResultsAlreadyExistsError(
+                ".".join([cluster_field, vector_fields[0], alias])
+            )
+        # load the documents
+        docs = self.get_all_documents(
+            dataset_id=dataset_id, filters=filters
+        )
+
+        # get vectors
+        if len(vector_fields) > 1:
+            raise ValueError(
+                "We currently do not support more than 1 vector field yet. This will be supported in the future."
+            )
+
+        if n_clusters is None and distance_threshold is None:
+            distance_threshold = 1
+        elif n_clusters is not None and distance_threshold is not None:
+            n_clusters = 10
```

Comment:
If we set a default, then we should add a warning

Reply:
`import warnings
warning.warning('setting distance_threshold=1`

like this?

Reply:
^ `logger` class might better but not sure if that is default on?